src/gesv_rbt.cc

```diff
@@ -0,0 +1,242 @@
+///    process
+/// - $\norm{r_j}_{inf}$ is the infinity-norm of the residual, $r_j = Ax_j - b_j$
+/// - $\norm{x_j}_{inf}$ is the infinity-norm of the solution
+/// - $\norm{A}_{inf}$ is the infinity-operator-norm of the matrix $A$
+///
+//------------------------------------------------------------------------------
+/// @tparam scalar_t
+///     One of float, double, std::complex<float>, std::complex<double>.
+//------------------------------------------------------------------------------
+/// @param[in,out] A
+///     On entry, the n-by-n matrix $A$ to be factored.
+///     On exit, the factors $L$ and $U$ from the factorization $A = P L U$;
+///     the unit diagonal elements of $L$ are not stored.
+///
+/// @param[out] pivots
+///     The pivot indices that define the permutation matrix $P$.
+///
+/// @param[in] B
+///     On entry, the n-by-nrhs right hand side matrix $B$.
+///
+/// @param[out] X
+///     On exit, if return value = 0, the n-by-nrhs solution matrix $X$.
+///
+/// @param[out] iter
+///     The number of the iterations in the iterative refinement
+///     process, needed for the convergence. If iterative refinement failed,
+///     it is set to -(1+itermax), regardless of whether the fallback solver
+///     was used.
+///
+/// @param[in] opts
+///     Additional options, as map of name = value pairs. Possible options:
+///     - Option::Lookahead:
+///       Number of panels to overlap with matrix updates.
+///       lookahead >= 0. Default 1.
+///     - Option::InnerBlocking:
+///       Inner blocking to use for panel. Default 16.
+///     - Option::Target:
+///       Implementation to target. Possible values:
+///       - HostTask:  OpenMP tasks on CPU host [default].
+///       - HostNest:  nested OpenMP parallel for loop on CPU host.
+///       - HostBatch: batched BLAS on CPU host.
+///       - Devices:   batched BLAS on GPU device.
+///     - Option::Depth:
+///       Depth for butterfly transform. Default 2
+///     - Option::Tolerance:
+///       Iterative refinement tolerance. Default epsilon * sqrt(m)
+///     - Option::MaxIterations:
+///       Maximum number of refinement iterations. Default 30
+///     - Option::UseFallbackSolver:
+///       If true and iterative refinement fails to convergene, the problem is
+///       resolved with partial-pivoted LU. Default true
+///
+/// TODO: return value
+/// @retval 0 successful exit
+/// @retval >0 for return value = $i$, the computed $U(i,i)$ is exactly zero.
+///         The factorization has been completed, but the factor U is exactly
+///         singular, so the solution could not be computed.
+///
+/// @ingroup gesv
+///
+template<typename scalar_t>
+void gesv_rbt(Matrix<scalar_t>& A,
+              Matrix<scalar_t>& B,
+              Matrix<scalar_t>& X,
+              int& iter,
+              Options const& opts)
+{
+    using real_t = blas::real_type<scalar_t>;
+
+    // gemmA and rbt are currently only implemented on the host
```

Comment:
I think gemmA is now working on device, right?

Reply:
Has `gemmA` been extended to multiple GPUs per process?  But, I should probably change it to call `gemm` on device and let the algorithm selection figure it out.